File: Sample/dump-load.py
import yaml

object1 = {
    'name': 'Ingy dot Net',
    'number': 42,
    'needs': [
        'sex',
        'drugs',
        'rock and roll'
    ]
}

yaml1 = yaml.dump(object1)
object2 = yaml.load(yaml1)
yaml2 = yaml.dump(object2)

if yaml1 == yaml2:
    print yaml1
else:
    raise Exception(yaml1 + "!=\n" + yaml2)

File: Sample/dump-load.rb
require 'yaml'

object1 = {
    'name' => 'Ingy dot Net',
    'number' => 42,
    'needs' => [
        'sex',
        'drugs',
        'rock and roll'
    ]
}

yaml1 = YAML::dump(object1)
object2 = YAML::load(yaml1)
yaml2 = YAML::dump(object2)

if yaml1 == yaml2
    print YAML.dump(object1)
else
    raise yaml1 + "!=\n" + yaml2
end
